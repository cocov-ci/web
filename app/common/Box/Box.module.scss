@import "styles/variables.module";

.box {
  background-color: $colorGrayLight;
  padding: 24px;
  border-radius: 8px;
  box-shadow: 0 4px 8px rgb(0 0 0 / 24%), 0 11px 40px rgb(0 0 0 / 24%);
  width: 100%;
  position: relative;
  overflow: hidden;

  &.gutterBottom {
    margin-bottom: 32px;
  }

  .content {
    display: flex;
    flex-direction: column;
    z-index: 1;
    position: relative;
  }

  .backgroundText {
    position: absolute;
    font-size: 141px;
    white-space: nowrap;
    opacity: 0.04;
    font-variation-settings: "wght" 700;
    line-height: 124px;
    left: 12px;
    bottom: -13px;

    &.small {
      font-size: 100px;
      line-height: 150px;
      bottom: -40px;
    }
  }
}
