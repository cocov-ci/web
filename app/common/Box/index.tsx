import classNames from 'classnames'
import React from 'react'

import styles from './Box.module.scss'

type BoxProps = {
  children?: React.ReactNode
  className?: string
  gutterBottom?: boolean
  backgroundTextSmall?: string
  backgroundTextLarge?: string
}

const Box = ({
  children,
  gutterBottom,
  backgroundTextLarge,
  backgroundTextSmall,
  className,
}: BoxProps) => {
  const backgroundText = backgroundTextLarge || backgroundTextSmall

  return (
    <div
      className={classNames(styles.box, className, {
        [styles.gutterBottom]: gutterBottom,
      })}
    >
      {backgroundText && (
        <span
          className={classNames(styles.backgroundText, {
            [styles.small]: Boolean(backgroundTextSmall),
          })}
        >
          {backgroundText}
        </span>
      )}
      <div className={styles.content}>{children}</div>
    </div>
  )
}

export default Box
