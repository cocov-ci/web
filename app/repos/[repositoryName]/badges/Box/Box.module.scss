@import "styles/variables.module";

.main {
  flex: 1;
  display: flex;
  padding-bottom: 30px;

  .box {
    flex: 1;

    &.extraPadding {
      padding-top: 70px;
    }
  }
}
