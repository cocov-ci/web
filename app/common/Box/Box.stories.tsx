import type { Meta, StoryObj } from '@storybook/react'

import Box from './index'

const meta: Meta<typeof Box> = {
  title: 'Common/Box',
  component: Box,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof Box>

export const Default: Story = {
  args: {
    children: <p>hey</p>,
  },
}
