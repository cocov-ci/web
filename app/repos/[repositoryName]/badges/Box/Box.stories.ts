import type { Meta, StoryObj } from '@storybook/react'

import Box from './index'

const meta: Meta<typeof Box> = {
  title: 'Badges/Box',
  component: Box,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof Box>

export const Default: Story = {
  args: {},
}
