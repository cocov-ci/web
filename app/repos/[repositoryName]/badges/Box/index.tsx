import Box from 'app/common/Box'

import styles from './Box.module.scss'

const BoxComponent = ({ children }: { children: React.ReactNode }) => {
  return (
    <div className={styles.main}>
      <Box className={styles.box}>{children}</Box>
    </div>
  )
}

export default BoxComponent
