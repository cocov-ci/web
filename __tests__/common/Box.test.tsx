import { render, screen } from '@testing-library/react'

import Box from 'app/common/Box'

import '@testing-library/jest-dom'

describe('Box', () => {
  const BoxComponent = () => (
    <Box>
      <p>children</p>
    </Box>
  )

  it('renders Box', () => {
    render(<BoxComponent />)

    expect(screen.getByText('children')).toBeVisible()
  })

  it('renders Box snapshots', () => {
    const { container } = render(<BoxComponent />)
    expect(container).toMatchSnapshot()
  })
})
