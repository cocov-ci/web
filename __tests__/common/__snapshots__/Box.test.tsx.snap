// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders Box snapshots 1`] = `
<div>
  <div
    class="box"
  >
    <p>
      children
    </p>
  </div>
</div>
`;
